MetaMask/metamask-extension
# feat: Update TransactionDetail Gas UI Styles

## **Description**

This PR updates various gas UI styles. There are discrepancies between transaction and gas details. This PR makes the styles more consistent, cleaner, and responsive. 

Figma: https://www.figma.com/design/oHNdOauz7w9PEti2zX83Cg/Gas-component?node-id=3-2837&t=wxI1mCX7Hf1TKfRk-0

It might be helpful to note that TransactionDetailITem is used in 13 various components
![CleanShot 2024-05-23 at 22 37 30](https://github.com/MetaMask/metamask-extension/assets/20778143/3496fe02-d1f6-442c-b368-80c57ce233ee)

There is a separate issue to update the gas UI for setApprovalForAll / revoke screens: https://github.com/MetaMask/metamask-extension/issues/24769

This was a meticulous operation as we had numerous combinations to account for with various styling means, including passing nested child components and the props suffixProps and textProps to styles. 

With the upcoming confirmation redesign work, we will eventually move away from these styles and TransactionDetail/TransactionDetailItem. In the meantime, we are updating the styles.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2194

## **Manual testing steps**

1. Go to Test Dapp
2. Test various transactions
3. Observe transaction and gas details

## **Screenshots/Recordings**



Note: My currency conversation API has been rate limited so double values are showing rather than Fiat + Native currency values

### **Before**

Some before screenshots can be found in the related Figma

<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/9cde3808-c339-4fb1-b49c-292f1eaf19f0" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/cfc630c6-b5ec-4644-b39e-6b9d8c4ebc6a" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/e5e88d60-8989-43ea-9dc4-5479f93ce25c" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/88a76c35-6482-46ad-b08f-aa7534285f40" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/326f96a7-c8f9-4adf-adaf-527bc4be426b" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/ca2e44de-05ef-438b-93be-e316023ae665" width="180">


### **After**

<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/6ebc6bb8-dfab-4587-b01d-5d5339409a85" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/9628cbb0-33b1-4365-a10c-90731f23f82a" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/72e1eb53-bcab-4a8f-89a2-f1b953778e88" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/66cfacd0-a3fa-41a8-8142-79b1226e188c" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/53ae16e6-d8b1-452b-a5f2-e22d69ea7f2d" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/19333cae-d3f1-4856-a890-9daaccc04ae8" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/2dfe5644-957b-4b91-92d8-e1aa1756e69a" width="180">


#### Layer 2
<img src="tps://github.com/MetaMask/metamask-extension/assets/20778143/67625848-a0ea-4f5f-91f4-f2c24f9cf989" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/19b3a4ab-d2e0-46a4-aa9a-75dd8f13bd6f" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/12bd9d04-1b7b-43ca-8dc6-31045961c044" width="320">
<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/bda88657-5c1e-41f1-a021-1cc5bcdcb600" width="320">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
